li-apache-kafka-clients/src/main/java/com/linkedin/kafka/clients/producer/LiKafkaInstrumentedProducerImpl.java

```diff
@@ -355,16 +353,20 @@ public void flush() {
    * If the underlying producer doesn't support a bounded flush, it will invoke the {@link #flush()}.
    */
   public void flush(long timeout, TimeUnit timeUnit) {
-    verifyOpen();
-
     delegateLock.readLock().lock();
     try {
       boolean useSeparateThreadForFlush = false;
       if (boundedFlushMethod != null) {
         try {
           boundedFlushMethod.invoke(delegate, timeout, timeUnit);
-        } catch (IllegalAccessException | InvocationTargetException e) {
-          throw new IllegalStateException("failed to invoke the bounded flush method", e);
+        } catch (IllegalAccessException illegalAccessException) {
+          throw new IllegalStateException("Failed to invoke the bounded flush method", illegalAccessException.getCause());
+        } catch (InvocationTargetException invocationTargetException) {
+          try {
+            throw (Exception) invocationTargetException.getCause();
+          } catch (Exception exception) {
```

Comment:
this could catch the exception thrown in line 366. 
One way to fix is to check if the "cause" is an instance of runtime exception
if the cause is a runtime exception, cast the exception to runtime exception and rethrow the exception.
if the cause is not a runtime exception, wrap the exception with the IllegalStateException as before.

Reply:
Hi Wesley, I have resolved this part according to your comments. Feel free to ping me if further modification is required. Thank you!